mlxtend/classifier/stacking_cv_classification.py

```diff
@@ -0,0 +1,270 @@
+                 use_features_in_secondary=False,
+                 stratify=True, verbose=0):
+
+        self.classifiers = classifiers
+        self.meta_classifier = meta_classifier
+        self.named_classifiers = {key: value for
+                                  key, value in
+                                  _name_estimators(classifiers)}
+        self.named_meta_classifier = {'meta-%s' % key: value for
+                                      key, value in
+                                      _name_estimators([meta_classifier])}
+        self.use_probas = use_probas
+        self.verbose = verbose
+        self.n_folds = n_folds
+        self.use_features_in_secondary = use_features_in_secondary
+        self.stratify = stratify
+
+    def fit(self, X, y):
+        """ Fit ensemble classifers and the meta-classifier.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Training vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+
+        """
+        self.clfs_ = [clone(clf) for clf in self.classifiers]
+        self.meta_clf_ = clone(self.meta_classifier)
+        if self.verbose > 0:
+            print("Fitting %d classifiers..." % (len(self.classifiers)))
+
+        if self.stratify:
+            skf = StratifiedKFold(y, n_folds=self.n_folds)
+        else:
+            skf = KFold(len(y), n_folds=self.n_folds)
+
+        all_model_predictions = np.array([]).reshape(len(y), 0)
+        for model in self.clfs_:
+
+            if self.verbose > 0:
+                i = self.clfs_.index(model) + 1
+                print("Fitting classifier%d: %s (%d/%d)" %
+                      (i, _name_estimators((model,))[0][0], i, len(self.clfs_)))
+
+            if self.verbose > 2:
+                if hasattr(model, 'verbose'):
+                    model.set_params(verbose=self.verbose - 2)
+
+            if self.verbose > 1:
+                print(_name_estimators((model,))[0][1])
+
+            if not self.use_probas:
+                single_model_prediction = np.array([]).reshape(0, 1)
+            else:
+                single_model_prediction = np.array([]).reshape(0, len(set(y)))
+
+            for num, (train_index, test_index) in enumerate(skf):
+
+                if self.verbose > 0:
+                    print ("Training and fitting fold %d of %d..." % \
+                          ((num+1), self.n_folds))
+
+                if not self.use_probas:
+                    prediction = model.fit(X[train_index], y[train_index])\
+                        .predict(X[test_index])
+                    prediction = prediction.reshape(prediction.shape[0], 1)
+                else:
+                    prediction = model.fit(X[train_index], y[train_index]).\
```

Comment:
This could be slightly refactored by fitting the model one time before the if else clause and then just call `model.predict` or `model.predict_proba` respectively